backend/applications.go

```diff
@@ -0,0 +1,257 @@
+// It will then return the pertinent information to the user.
+func (ar *ApplicationsRouter) newApp(w http.ResponseWriter, r *http.Request) {
+	ctx := context.Background()
+	// Create a new Application struct
+	app := application{}
+	// parse the request body into the registration variable
+	app.apiKey = ar.apiKeyGenerate()
+	app.appName = r.FormValue("title")
+	app.description = r.FormValue("description")
+	app.owners = r.FormValue("owners")
+	app.teamName = r.FormValue("teamName")
+	app.appID = ar.appIDGenerate()
+	app.isValid = true
+
+	/* Storing information in the Database*/
+
+	// Initiating a transaction
+	tx, err := ar.Svcs.Db.Begin(ctx)
+	if err != nil {
+		ar.Svcs.Logger.Error(err)
+		ar.Svcs.Logger.Info("Error initiating database transaction")
+		return
+	}
+	defer tx.Rollback(ctx)
+
+	// Getting prepared statements ready to use later
+	keyStmt, err := tx.Prepare(ctx, "insert_into_keys", "INSERT INTO keys (id, apiKey, isValid) VALUES ($1, $2, $3)")
+	if err != nil {
+		ar.Svcs.Logger.Error(err)
+		ar.Svcs.Logger.Info("Error preparing database statement")
+		return
+	}
+	appStmt, err := tx.Prepare(ctx, "insert_into_applications", "INSERT INTO applications (id, appName, description, owners, teamName) VALUES ($1, $2, $3, $4, $5)")
+	if err != nil {
+		ar.Svcs.Logger.Error(err)
+		ar.Svcs.Logger.Info("Error preparing database statement")
+		return
+	}
+
+	// Storing info in the keys table
+	_, err = keyStmt.Exec(ctx, app.appID, app.apiKey, app.isValid)
+	if err != nil {
+		ar.Svcs.Logger.Error(err)
+		ar.Svcs.Logger.Info("Error storing new app info in database (keys table)")
+		return
+	}
+
+	// Storing info in the applications table
+	query2 := `INSERT INTO applications (id, appName, description, owners, teamName) VALUES ($1, $2, $3, $4, $5)`
+	_, err = tx.Exec(ctx, query2, app.appID, app.appName, app.description, app.owners, app.teamName)
+	if err != nil {
+		ar.Svcs.Logger.Error(err)
+		ar.Svcs.Logger.Info("Error storing new app info in database (applications table)")
+		return
+	}
+
+	// Commit the transaction
+	err = tx.Commit(ctx)
+	if err != nil {
+		ar.Svcs.Logger.Error(err)
+		ar.Svcs.Logger.Info("Error committing database transaction")
+		return
+	}
+
+}
+
+// The applications/myApps endpoint.
+// This function will return a list of all the applications that the user owns.
+func (ar *ApplicationsRouter) myApps(w http.ResponseWriter, r *http.Request) {
+	ctx := context.Background()
+	// How do we get the user's email?
+	query := "SELECT appName, appID, description, owners FROM applications WHERE owners=$1" // Need to fix this so that it checks if the string in the owners column contains the user's email
+	rows, err := ar.Svcs.Db.Query(ctx, query, "email")
+	if err != nil {
+		ar.Svcs.Logger.Error(err)
+		ar.Svcs.Logger.Info("Error querying database for user's applications")
+		return
+	}
+	// How do I package this information into the response?
+
```

Comment:
Here, we could create an empty array of Applications, scan each row into a new Application and append it to the array, then leveraging the struct tags [encode the array](https://github.com/elonsoc/ods/blob/390054fdfc361f94d74fc98c753bb862cc3bf997/backend/locations/v1/buildings/buildings.go#L114-L115) in JSON and write to the ResponseWriter.

Reply:
This sounds like a good plan. I have got it working on my testing psql server, but I have hit a road block when it comes to coding. My issue is that I dont have the users email. I know @jumarmartin said it would be passed through like the SAML assertion and stored locally or something like that, but for right now I do not understand how to get that info. This prevents me really from completing the other endpoints. I will try to code around this sometime this week.

Reply:
REBASE: DONE
ACCESSING SAML INFO: NOT DONE

Dummy endpoints have been set up, I will access saml to correct the endpoint at a later date.